src/Tools/DynamoInstallDetective/AscSDKWrapper.cs

```diff
@@ -0,0 +1,154 @@
+
+                if(status != ASC_STATUS.SUCCESS)
+                {
+                    fetchFromEnv = false;
+                }
+            }
+
+            return status;
+        }
+
+        private ASC_STATUS ReadASCInstallPathFromRegistry(string majorRelease)
+        {
+            ASC_STATUS status = ASC_STATUS.REG_FAILED;
+
+            var registryPath = Path.Combine(BaseKey.Name, registryKey, majorRelease);
+            if(string.IsNullOrEmpty(regPath))
+            {
+                regPath = registryPath;
+            }
+
+            const string valueName = @"InstallPath";
+            const string versionName = @"Version";
+
+            string installPath = Registry.GetValue(registryPath, valueName, null) as String;
+            if (installPath != null)
+            {
+                if (!Directory.Exists(installPath))
+                {
+                    return ASC_STATUS.INCORRECT_REG_PATH;
+                }
+
+                if (fetchFromEnv)
+                {
+                    this.installPath = installPath;
+                    status = ASC_STATUS.SUCCESS;
+                }
+                else
+                {
+                    string version = Registry.GetValue(registryPath, versionName, null) as String;
+                    if (string.IsNullOrEmpty(version))
+                    {
+                        return ASC_STATUS.EMPTY_REG_VERSION;
+                    }
+
+                    this.installPath = Path.Combine(installPath, version);
+
+                    if (!Directory.Exists(this.installPath))
+                    {
+                        return ASC_STATUS.INCORRECT_REG_PATH;
+                    }
+                    status = ASC_STATUS.SUCCESS;
+                }
+            }
+            return status;
+        }
+        /// <summary>
+        /// Initialize ASC wrapper
+        /// </summary>
+        /// <param name="release">Major ASC release, for example 2026</param>
+        public AscSdkWrapper(string release)
+        {
+            majorRelease = release;
+            version = @"ACS_VERSION_" + majorRelease;
+        }
+
+        /// <summary>
+        /// Get the install path for the major release
+        /// </summary>
+        /// <param name="installedPath">The install path for the major release if successful</param>
+        /// <returns>ASC_STATUS.SUCCESS if path was retrived successfully</returns>
+        public ASC_STATUS GetInstalledPath(ref string installedPath)
+        {
+            if(string.IsNullOrEmpty(majorRelease))
+            {
+                return ASC_STATUS.INITIALIZE_FAILED;
+            }
+
+            var status = ReadASCVersionFromEnv() == ASC_STATUS.SUCCESS ? ASC_STATUS.SUCCESS : ReadASCInstallPathFromRegistry(majorRelease);
+            installedPath = installPath;
+
+            return status;
+        }
+
+        /// <summary>
+        /// Get the major version of all ASC packages installed on the local machine
```

Comment:
Are these more like version informations including the package/ component names and other properties?